lib/telegraf.py

```diff
@@ -0,0 +1,199 @@
+#
+# [EXPIRIMENTAL]
+#
+# Add Telegraf to an app container to collect StatsD events from the runtime.
+# Metrics will be forwarded to host defined in APPMETRICS_TARGET environment
+# variable which is a JSON (single or array) with the following values
+# - url: complate url of the endpoint. Mandatory.
+# - username: basic auth username. Optional.
+# - password: basic auth password. Mandatory if username is specified.
+# - kpionly: indicates that by default only metrics with KPI=true tag are passed
+#     to custom end points
+#
+# Examples:
+# {"url: "https://customreceiver.mydomain.com", "username": "user", "password": secret", "kpionly": true}
+# [{"url: "https://customreceiver.mydomain.com", "username": "user", "password": secret", "kpionly": true}]
+#
+
+import json
+import os
+
+import buildpackutil
+import datadog
+import subprocess
+from m2ee import logger
+
+
+def _get_appmetrics_target():
+    return os.getenv('APPMETRICS_TARGET')
+
+
+def is_enabled():
+    return _get_appmetrics_target() is not None
+
+
+def _get_tags():
+    # Telegraf tags must be key / value
+    tags = {}
+    for kv in [t.split(':') for t in buildpackutil.get_tags()]:
+        if len(kv) == 2:
+            tags[kv[0]] = kv[1]
+        else:
+            logger.warn('Skipping tag % because not a key/value')
+    return tags
+
+
+def _config_value_str(value):
+    if type(value) is str:
+        return '"%s"' % value
+    elif type(value) is int:
+        return value
+    elif type(value) is bool:
+        return str(value).lower()
+    elif type(value) is list:
+        return json.dumps(value)
+
+
+def _create_config_file(agent_config):
+    logger.info('writing config file')
```

Comment:
Log as `debug` is preferred for this kind of cases to prevent excessively logging. Please revisit the other logging calls also.

Reply:
done